content/terraform/v1.12.x/docs/language/block/stack/deploy/deployment.mdx

```diff
@@ -0,0 +1,294 @@
+- Use the [`deployment_auto_approve`](/terraform/language/block/stack/deploy/deployment-auto-group) block to define rules that automatically approve deployment plans based on specific conditions.
+- Use the [`identity_token`](/terraform/language/block/stack/deploy/identity_token) block to authenticate a deployment with a provider configuration using OIDC.
+- Use the [`store`](/terraform/language/block/stack/deploy/store) block to define key-value secrets in your deployment configuration and use existing variable sets in HCP Terraform.
+- Use the [`locals`](/terraform/language/block/stack/deploy/locals) block to define local values that you can reuse within your deployment configuration.
+- Use the [`publish_output`](/terraform/language/block/stack/deploy/publish_output) block to output values that you want to be available to other Stacks.
+- Use the [`upstream_input`](/terraform/language/block/stack/deploy/upstream_input) block to source inputs from another Stack.
+
+
+## Configuration model
+
+The `deployment` block supports the following arguments:
+
+- [`deployment "<LABEL>"`](#deployment) &nbsp; block
+  - [`inputs`](#inputs) &nbsp; map
+  - [`deployment_group`](#deployment_group) &nbsp; reference to deployment group
+  - [`destroy`](#destroy) &nbsp; boolean
+
+## Complete configuration
+
+All available arguments are defined in the following `deployment` block:
+
+```hcl
+deployment "<LABEL>" {
+  inputs = {
+    <VARIABLE_NAME> = <VALUE>
+  }
+  deployment_group = "deployment_group.<NAME>"
+  destroy = <BOOLEAN>
+}
+```
+
+## Specification
+
+A `deployment` block supports the following configuration.
+
+### `deployment "<LABEL>"`
+
+The label after the `deployment` keyword is a name for the deployment, which must be unique among all deployments in the same deployment configuration file. The name of a deployment can be any valid [identifier](/terraform/language/syntax/configuration#identifiers).
+
+The `deployment` block supports the following arguments:
+
+| Argument | Description | Type | Required? |
+| --- | --- | --- | --- |
+| `inputs` | A mapping of Stack variable names to values for this deployment. | Map | Required |
+| `deployment_group` | A reference to the group that this deployment belongs to. | Reference | Optional |
+| `destroy` | A boolean flag that indicates whether HCP Terraform should destroy this deployment. | Boolean | Optional |
+
+### `inputs`
+
+The `inputs` argument defines a mapping of Stack variable names to values for this specific deployment.
+
+```hcl
+deployment "<LABEL>" {
+  inputs = {
+    <VARIABLE_NAME> = <VALUE>
+    <VARIABLE_NAME_TWO> = <VALUE_TWO>
+  }
+}
+```
+
+The keys in the `inputs` map must correspond to the names of the variables defined in your component configuration, and the values must be valid HCL literals that meet the type constraints of those variables.
```

Comment:
Doesn't have to be literals, you can use expressions here as well! That's how you're able to reference locals, identity tokens, and values from store objects. 

```suggestion
The keys in the `inputs` map must correspond to the names of the variables defined in your component configuration, and the values must be valid HCL expressions that meet the type constraints of those variables. Input value expressions can reference `locals`, `identity_token`, `store`, and `upstream_value` blocks. 
```